contracts/src/token/erc721/extensions/consecutive.rs

```diff
@@ -0,0 +1,1079 @@
+                    maxBatch: U256::from(MAX_BATCH_SIZE),
+                }
+                .into());
+            }
+
+            let last = next + batch_size - uint!(1_U96);
+            self._sequential_ownership
+                .push(last, U160::from_be_bytes(to.into_array()))?;
+
+            self.erc721._increase_balance(to, U128::from(batch_size));
+            evm::log(ConsecutiveTransfer {
+                fromTokenId: next.to::<U256>(),
+                toTokenId: last.to::<U256>(),
+                fromAddress: Address::ZERO,
+                toAddress: to,
+            });
+        };
+        Ok(next)
+    }
+
+    /// Should be called to restrict consecutive mint after.
+    /// After this function being called, every call to
+    /// [`Self::_mint_consecutive`] will fail.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Write access to the contract's state.
+    pub fn _stop_mint_consecutive(&mut self) {
+        self._initialized.set(true);
+    }
+
+    /// Override of [`Erc721::_update`] that restricts normal minting to after
+    /// construction.
+    ///
+    /// WARNING: Using [`Erc721Consecutive`] prevents minting during
+    /// construction in favor of [`Erc721Consecutive::_mint_consecutive`].
+    /// After construction,[`Erc721Consecutive::_mint_consecutive`] is no
+    /// longer available and minting through [`Erc721Consecutive::_update`]
+    /// becomes possible.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `to` - Account of the recipient.
+    /// * `token_id` - Token id as a number.
+    /// * `auth` - Account used for authorization of the update.
+    ///
+    /// # Errors
+    ///
+    /// If token does not exist and `auth` is not `Address::ZERO`, then the
+    /// error [`erc721::Error::NonexistentToken`] is returned.
+    /// If `auth` is not `Address::ZERO` and `auth` does not have a right to
+    /// approve this token, then the error
+    /// [`erc721::Error::InsufficientApproval`] is returned.
+    /// If consecutive mint wasn't finished yet (function
+    /// [`Self::_stop_mint_consecutive`] wasn't called) error
+    /// [`Error::ERC721ForbiddenMint`] is returned.
+    ///
+    /// # Events
+    ///
+    /// Emits a [`Transfer`] event.
+    pub fn _update(
+        &mut self,
+        to: Address,
+        token_id: U256,
+        auth: Address,
+    ) -> Result<Address, Error> {
+        let previous_owner = self.__update(to, token_id, auth)?;
+
+        // only mint after construction
+        if previous_owner == Address::ZERO && !self._initialized.get() {
+            return Err(ERC721ForbiddenMint {}.into());
+        }
+
+        // record burn
+        if to == Address::ZERO // if we burn
+            && token_id < U256::from(self._next_consecutive_id()) // and the tokenId was minted in a batch
+            && !self._sequentian_burn.get(token_id)
+        // and the token was never marked as burnt
```

Comment:
```suggestion
        let next_id = self._next_consecutive_id();
        let is_burn = to == Address::ZERO;
        let token_minted_in_batch = token_id < U256::from(next_id);
        let not_marked_burned = !self._sequentian_burn.get(token_id);
        // Record burn.
        if is_burn && token_minted_in_batch && not_marked_burned {
            self._sequentian_burn.set(token_id);
        }
```

Reply:
The other solution could be: Create small functions with corresponding names that will be lazily evaluated. Or leave it as it is right now (like in solidity).

Reply:
Ah, you are totally right. Yeah, in this case I think being more performant is better. So let's just ignore 👍